api/src/main/java/ca/bc/gov/educ/api/gradstudent/messaging/MessageSubscriber.java

```diff
@@ -0,0 +1,91 @@
+package ca.bc.gov.educ.api.gradstudent.messaging;
+
+import ca.bc.gov.educ.api.gradstudent.model.dc.Event;
+import ca.bc.gov.educ.api.gradstudent.orchestrator.base.EventHandler;
+import ca.bc.gov.educ.api.gradstudent.service.events.EventHandlerDelegatorService;
+import ca.bc.gov.educ.api.gradstudent.util.EducGradStudentApiConstants;
+import ca.bc.gov.educ.api.gradstudent.util.JsonUtil;
+import ca.bc.gov.educ.api.gradstudent.util.LogHelper;
+import io.nats.client.Connection;
+import io.nats.client.Message;
+import io.nats.client.MessageHandler;
+import jakarta.annotation.PostConstruct;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static ca.bc.gov.educ.api.gradstudent.constant.Topics.GRAD_STUDENT_API_TOPIC;
+import static lombok.AccessLevel.PRIVATE;
+
+@Component
+@Slf4j
+public class MessageSubscriber {
+
+  @Getter(PRIVATE)
+  private final Map<String, EventHandler> handlerMap = new HashMap<>();
+  @Getter(PRIVATE)
+  private final EventHandlerDelegatorService eventHandlerDelegatorService;
+  private final Connection connection;
+  private final EducGradStudentApiConstants constants;
+
+  @Autowired
+  public MessageSubscriber(final Connection con, final List<EventHandler> eventHandlers, final EducGradStudentApiConstants constants, final EventHandlerDelegatorService eventHandlerDelegatorService) {
+    this.eventHandlerDelegatorService = eventHandlerDelegatorService;
+    this.connection = con;
+    eventHandlers.forEach(handler -> {
+      this.handlerMap.put(handler.getTopicToSubscribe(), handler);
+      this.subscribeForSAGA(handler.getTopicToSubscribe(), handler);
```

Comment:
I see some redundant codes in this class.
handlerMap is to handle multiple subscribers.  I think subscribe() method below could be enhanced to use handlerMap so that onMessage could handle messages from multiple topics instead of having the separate onMessageForSAGA.